src/hypothesis/core.py

```diff
@@ -956,6 +956,17 @@ def run_test_with_generator(test):
 
         original_argspec = getfullargspec(test)
 
+        if getattr(test, 'is_hypothesis_test', False):
+            note_deprecation((
+                'You have applied @given to a test more than once. In future '
+                'this will be an error. Applying @given twice wraps the test '
+                'twice, which can be extremely slow. A similar effect can be '
+                'gained by combining the arguments to the two calls to '
+                'given. For example, instead of @given(booleans()) '
+                '@given(integers()), you could write @given(booleans(), '
```

Comment:
`@given` actually fills positional arguments from the right, so the order would be reversed!

Reply:
I thought that too, but then I double checked and this way around is actually correct. The inner given fills the rightmost position, then the outer given fills the one after that - i.e. the one to its left.

Reply:
...so it does.  *facepalm*

Reply:
I confess I spent a while wondering about the order things would come out in, and in the end just tested to see what happened...